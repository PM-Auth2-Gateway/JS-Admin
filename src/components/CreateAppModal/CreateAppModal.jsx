import React, { useState } from 'react';
import { Button, Modal, Form } from 'react-bootstrap';

const CreateAppModal = () => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const submitForm = () => {
    handleClose();
  };

  return (
    <React.Fragment>
      <Button data-test={'open-modal'} variant="danger" onClick={handleShow}>
        Create Application
      </Button>

      <Modal show={show} onHide={handleClose} centered>
        <Modal.Header closeButton>
          <Modal.Title>Create Application</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form.Group>
            <Form.Label>Name</Form.Label>
            <Form.Control type="text" placeholder="Enter name of app" />
            <Form.Text className="text-muted">
              You can change the application name later in the application
              settings.
            </Form.Text>
          </Form.Group>
        </Modal.Body>

        <Modal.Footer>
          <Button
            data-test={'close-modal'}
            variant="outline-secondary"
            onClick={handleClose}
          >
            Cancel
          </Button>
          <Button
            data-test={'submit-modal'}
            variant="primary"
            type="submit"
            onClick={submitForm}
          >
            Create
          </Button>
        </Modal.Footer>
      </Modal>
    </React.Fragment>
  );
};

export default CreateAppModal;
