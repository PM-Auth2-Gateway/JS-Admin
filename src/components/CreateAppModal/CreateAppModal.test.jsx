import React, { useState as useStateMock } from 'react';
import { useDispatch } from 'react-redux';
import { shallow } from 'enzyme';

import CreateAppModal from './CreateAppModal';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

describe('CreateAppModal Component', () => {
  const setState = jest.fn();
  const dispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);
    useStateMock.mockImplementation((init) => [init, setState]);
  });

  it('should render CreateAppModal Component', function () {
    const component = shallow(<CreateAppModal />);

    expect(component).toMatchSnapshot();
  });

  it('should open modal', function () {
    const component = shallow(<CreateAppModal />);

    const openModalBtn = component.find(`[data-test="open-modal"]`);

    openModalBtn.simulate('click');
  });
});
