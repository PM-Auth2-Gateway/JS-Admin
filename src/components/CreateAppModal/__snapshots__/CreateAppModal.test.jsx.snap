// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAppModal Component should render CreateAppModal Component 1`] = `
<Fragment>
  <Button
    active={false}
    data-test="open-modal"
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    Create Application
  </Button>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
  >
    <Formik
      initialValues={
        Object {
          "name": "",
        }
      }
      onSubmit={[Function]}
      validationSchema={
        ObjectSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_excludedEdges": Array [],
          "_nodes": Array [
            "name",
          ],
          "_sortErrors": [Function],
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {},
          "fields": Object {
            "name": StringSchema {
              "_blacklist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_blacklistError": undefined,
              "_mutate": undefined,
              "_typeError": [Function],
              "_whitelist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_whitelistError": undefined,
              "conditions": Array [],
              "deps": Array [],
              "exclusiveTests": Object {
                "max": true,
                "min": true,
                "required": true,
              },
              "spec": Object {
                "abortEarly": true,
                "nullable": false,
                "presence": "required",
                "recursive": true,
                "strict": false,
                "strip": false,
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "optional",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
          "type": "object",
        }
      }
    >
      <Component />
    </Formik>
  </Modal>
</Fragment>
`;
